pages/tokens.py

```diff
@@ -96,6 +110,42 @@
     ]
 )
 
+#Update the 'selected-rows' store based on changes in selected rows in 'tokens-table'
+@callback(
+    Output('selected-rows', 'data'),
+    Input('tokens-table', 'selected_rows'),
+    prevent_initial_call=True
+)
+def update_selected_rows(selected_rows):
+    #Update the 'selected-rows' store with the current selected rows
+    return selected_rows
+
+
+@callback(
+    [Output('tokens-table', 'data'),
+    Output('tokens-table', 'selected_rows')],
+    Input('confirm-delete','submit_n_clicks'),
+    State('tokens-table', 'data'),
+    State('selected-rows', 'data'),
+    prevent_initial_call=True
+)
+#Delete selected rows based on confirmation
+def delete_selected_rows(submit_clicks, current_data, selected_rows):
+    #Check if the delete confirmation button is clicked
+    if submit_clicks and token_deletion_enabled:
+        #Remove selected rows from the current data
+        current_data = [row for i, row in enumerate(current_data) if i not in selected_rows]
```

Comment:
cleanup fix: I would suggest re-reading the table from the database instead of making local changes here. The admin dashboard runs on the server, so we do not anticipate network drops between the python code and the DB. And this will ensure that if there are issues with deleting the data, we don't show a wrong view to the user. It will also handle consistency between parallel sessions (e.g. another admin is creating tokens at the same time).

If we do want to keep this code (which I don't recommend), we have to make the local changes after the DB changes, and use a try/catch block to ensure that the local changes only happen if the DB changes were successful.